### PR TITLE
Honor reorient and quit request from OVR

### PR DESCRIPTION
**Pull request by Atlante45**
Respond the reset orientation and quit calls from Oculus

Test Plan:
While in the Oculus:
- Turn around (Away from the UI)
- Open the Oculus menu (Power button on Xbox controller)
- Choose to reorient
- Make sure UI is back in front of you.
- Open the Oculus menu
- Exit to Oculus home
- Make sure Interface shuts down.

### CODE REVIEW

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4750,7 +4750,7 @@ void Application::resetSensors(bool andReload) {
     DependencyManager::get<EyeTracker>()->reset();
     getActiveDisplayPlugin()->resetSensors();
     _overlayConductor.centerUI();
-    getMyAvatar()->reset(andReload);
+    getMyAvatar()->reset(true, andReload);
```
Comment (Atlante45) [inline interface/src/Application.cpp]: Is this a bug?! I've seen that while looking through this code and the argument name does not match the signature.

File: plugins/oculus/src/OculusBaseDisplayPlugin.cpp [hunk 2]
```diff
@@ -21,6 +21,15 @@ void OculusBaseDisplayPlugin::resetSensors() {
 }
 
 bool OculusBaseDisplayPlugin::beginFrameRender(uint32_t frameIndex) {
+    handleOVREvents();
+    if (quitRequested()) {
+        QMetaObject::invokeMethod(qApp, "quit");
+        return false;
```
Comment (birarda) [inline plugins/oculus/src/OculusBaseDisplayPlugin.cpp]: if we return here are we going to break rendering and possibly create an uncomfortable situation while the application is quitting?

Comment (Atlante45) [inline plugins/oculus/src/OculusBaseDisplayPlugin.cpp]: No, Oculus takes over and puts you in the "Oculus menu".
We do the same with  SteamVR.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7b0b7770513933f68d64ea19aa00a58654c807bc available at: https://gist.github.com/ac019a8f8a1e59a765e52287263f17e6<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9167-7b0b777.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9167-7b0b777.dmg

**Comment by claudioshg** [comment]
- Downloaded PR-9167.exe, and Installed
- Tested on Home and Welcome rooms
- Turned around and went into Oculus Home Screen by pressing the Home button on the Xbox controller. 
- Chose option to orientate
- Made sure HUD reappeared in front of me
- Went back into Oculus Home Screen and exited to home.
- App failed to close

**Comment by Atlante45** [comment]
@claudioshg Did you get any specific error message? I just tried with the PR build and it worked fine for me.

**Comment by claudioshg** [comment]
appfail

This is the error message I get

**Comment by Atlante45** [comment]
Any chance Interface had a modal window open?

**Comment by claudioshg** [comment]
What is a modal window?

**Comment by claudioshg** [comment]
Retried testing again.
First try- The program froze right after accepting to reorientate. Stuck on the spinner.
Second try- Passed
Third try- Application failed to close again.

**Comment by birarda** [comment]
We should definitely investigate into why @claudioshg is having issues but I tested the "Exit to Home" and "Reset View in App" features and they both worked as expected.

File: plugins/oculus/src/OculusBaseDisplayPlugin.cpp [hunk 3]
```diff
@@ -21,6 +21,15 @@ void OculusBaseDisplayPlugin::resetSensors() {
...
+    }
+    if (reorientRequested()) {
+        emit resetSensorsRequested();
```
Comment (jherico) [inline plugins/oculus/src/OculusBaseDisplayPlugin.cpp]: Once you emit this you need to call `ovr_ClearShouldRecenterFlag`

**Comment by kevinmthomasse** [comment]
testing

**Comment by cozza13** [comment]
I crashed exiting

**Pull request merged**